scrapy/utils/ftp.py

```diff
@@ -13,3 +15,20 @@ def ftp_makedirs_cwd(ftp, path, first_call=True):
         ftp.mkd(path)
         if first_call:
             ftp.cwd(path)
+
+def ftp_store_file( 
+        path, file, host ,port,
+        username, password, use_active_mode=False):
+    """Opens a FTP connection with passed credentials,sets current directory
+    to the directory extracted from given path, then uploads the file to server
+    """
+    ftp = FTP()
```

Comment:
I think it can be better to use context manager here (see https://docs.python.org/3/library/ftplib.html#ftplib.FTP); this may allow to remove some code, and handle errors in a more correct way.

Reply:
For error handling, for now exceptions are just thrown, do you suggest creating new custom exceptions and throw them instead ? If this is not the case, I don't think it will make a difference. I can change it though.

Reply:
I mean use 

```py
with FTP(host, user=username, passwd=password) as ftp:
    if use_active_mode:
        ftp.set_pasv(False)
    # ...   
```
This would ensure ftp.quit is called even if some step in the middle fails.

But now, as I'm looking at it, it is not clear how to pass `port` argument this way. No idea why isn't it a part of api.

Reply:
Yes, the `FTP` constructor has no `port` parameter, it uses the default port, but the `connect` function has `port` parameter, so might use a work around like
```
with FTP() as ftp:
    ftp.connect(host, port)
    ftp.login(username, password)
    # ...
```
Not much code will be removed then, but port could be changed and `quit` would always be called. what do you think?

Reply:
sounds good @OmarFarrag! Could you please also check merge conflicts?

Reply:
Okay on it